x/evm/config/denom_config.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+//
+// The config package provides a convinient way to modify x/evm params and values.
+// Its primary purpose is to be used during application initialization.
+
+package config
+
+type Decimals uint32
+
+const (
+	SixDecimals      Decimals = 6
+	EighteenDecimals Decimals = 18
+)
+
+// EvmDenom struct holds the name and decimals of the EVM denom
+type EvmDenom struct {
+	denom    string
+	decimals Decimals
+}
+
+var evmDenom EvmDenom
+
+func SetDecimals(d Decimals) {
```

Comment:
please add a seal and the function comments

Reply:
The sealing happens after calling the configure method of the EVMConfigurator. Calling sealing for every setter of the struct fields is a bit redundant and break the builder pattern. It was a design choice of the configurator.

Reply:
Setters are now private so they cannot be called if not via the EVMConfigurator type. The only way to change them multiple type is instantiating more than once this struct, which a conscious error of the dev.